ch_backup/util.py

```diff
@@ -369,3 +369,12 @@ def dataclass_from_dict(type_: Type[T], data: dict) -> T:
     """
     class_fields = {f.name for f in data_fields(type_)}
     return type_(**{k: v for k, v in data.items() if k in class_fields})  # type: ignore[call-arg]
+
+
+def cached_property(f):
```

Comment:
How long will this cache properties? What if some properties change, how to invalidate cache?

Reply:
I no longer like this approach, although I had previously advised it. It does not fit this case very well.

It caches forever and can lead to memory leaks when creating a large number of objects. It will keep these objects alive.

I propose to ged rid of generalization and use simple ad hoc property with saving context in _variable. And setting it to None when clear is needed.

Or [this](https://github.com/bottlepy/bottle/blob/3cc41c83545d43d3577bf88cd160dd64ab146359/bottle.py#L175) approach which is more suitable for this case. For invalidating it is used `del self.my_cached_attribute`.

Reply:
Fixed it. Made it as Alexey  proposed with cached_property and invalidating to avoid memory leaks.
>Or [this](https://github.com/bottlepy/bottle/blob/3cc41c83545d43d3577bf88cd160dd64ab146359/bottle.py#L175) approach which is more suitable for this case than current one. For invalidating it is used del self.my_cached_attribute.